packages/forms-web-app/src/dynamic-forms/dynamic-components/add-more/question.js

```diff
@@ -0,0 +1,217 @@
+
+		// get viewModel for add more component
+		if (this.#hasAtLeastOneAnswer(answers)) {
+			const viewModel = super.prepQuestionForRendering(section, journey, customViewData);
+			viewModel.addMoreAnswers = this.#addListingDataToViewModel(journey);
+			return viewModel;
+		}
+
+		// get viewModel for subQuestion
+		return this.subQuestion.prepQuestionForRendering(section, journey, customViewData);
+	}
+
+	/**
+	 * @param {Journey} journey
+	 * @returns {Array.<addMoreAnswer>} list of existing answers to the sub question
+	 */
+	#addListingDataToViewModel(journey) {
+		const addMoreAnswers = [];
+
+		const answers = journey.response.answers[this.fieldName];
+
+		let i = 1;
+		for (const item in answers) {
+			const answer = answers[item];
+			addMoreAnswers.push({
+				label: `${this.subQuestionLabel} ${i}`,
+				answer: this.subQuestion.formatAnswerForSummary(answer[this.subQuestion.fieldName]),
+				removeLink: '#' + answer.addMoreId
+			});
+			i++;
+		}
+
+		return addMoreAnswers;
+	}
+
+	/**
+	 * Takes the data to save from the subquestion and nests with property name
+	 * @param {ExpressRequest} req
+	 * @param {JourneyResponse} journeyResponse - current journey response, modified with the new answers
+	 * @returns {Promise.<Object>}
+	 */
+	async getDataToSave(req, journeyResponse) {
+		// get current answers
+		let responseToSave = {
+			answers: {
+				[this.fieldName]: []
+			}
+		};
+
+		if (journeyResponse.answers[this.fieldName]) {
+			responseToSave.answers[this.fieldName] = [...journeyResponse.answers[this.fieldName]];
+		}
+
+		// get answer to subquestion
+		const individual = await this.subQuestion.getDataToSave(req, journeyResponse);
+
+		responseToSave.answers[this.fieldName].push({
+			...individual.answers,
+			addMoreId: uuid.v4()
+		});
+
+		return responseToSave;
+	}
+
+	/**
+	 * Handles redirect after saving, add more questions loop back to themselves
+	 * @param {ExpressResponse} res
+	 * @param {Journey} journey
+	 * @param {string} sectionSegment
+	 * @param {string} questionSegment
+	 * @returns {void}
+	 */
+	handleNextQuestion(res, journey, sectionSegment, questionSegment) {
+		return res.redirect(journey.getCurrentQuestionUrl(sectionSegment, questionSegment));
+	}
+
+	/**
+	 * Save an uploaded file
+	 * @param {ExpressRequest} req
+	 * @param {ExpressResponse} res
+	 * @param {Journey} journey
+	 * @param {Section} section
+	 * @param {JourneyResponse} journeyResponse
+	 * @returns {Promise<void>}
+	 */
+	saveAction = async (req, res, journey, section, journeyResponse) => {
+		const isAddMorePage = req.body['add-more-question'];
+		const addMoreAnswer = req.body[this.fieldName];
+
+		if (isAddMorePage) {
+			const errorViewModel = this.checkForValidationErrors(req, section, journey);
+			if (errorViewModel) {
+				return this.renderAction(res, errorViewModel);
+			}
+
+			if (addMoreAnswer === 'yes') {
+				return this.renderAction(res, this.subQuestion.prepQuestionForRendering(section, journey));
```

Comment:
Do we need different urls for the subquestion?
If so how best to achieve that?

Reply:
I wouldn't have thought we'd need a separate URL as we don't have a scenario where we'd want the user to be able to directly access a sub-question.